python/tests/io/test_fsspec.py

```diff
@@ -0,0 +1,191 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import uuid
+
+import pytest
+
+from pyiceberg.io import fsspec
+
+
+@pytest.mark.s3
+def test_fsspec_new_input_file(fsspec_fileio):
+    """Test creating a new input file from an fsspec file-io"""
+    filename = str(uuid.uuid4())
+
+    input_file = fsspec_fileio.new_input(f"s3://testbucket/{filename}")
+
+    assert isinstance(input_file, fsspec.FsspecInputFile)
+    assert input_file.location == f"s3://testbucket/{filename}"
+
+
+@pytest.mark.s3
+def test_fsspec_new_s3_output_file(fsspec_fileio):
+    """Test creating a new output file from an fsspec file-io"""
+    filename = str(uuid.uuid4())
+
+    output_file = fsspec_fileio.new_output(f"s3://testbucket/{filename}")
+
+    assert isinstance(output_file, fsspec.FsspecOutputFile)
+    assert output_file.location == f"s3://testbucket/{filename}"
+
+
+@pytest.mark.s3
+def test_fsspec_write_and_read_file(fsspec_fileio):
+    """Test writing and reading a file using FsspecInputFile and FsspecOutputFile"""
+    filename = str(uuid.uuid4())
+    output_file = fsspec_fileio.new_output(location=f"s3://testbucket/{filename}")
+    f = output_file.create()
```

Comment:
Can you use the context manager in the tests? Just to give a good example:
```python
with output_file.create() as f:
    f.write(b"foo")
```

Reply:
Updated! Changed all the tests to use context managers. Thanks @Fokko